ragna/deploy/_cli/core.py

```diff
@@ -171,3 +177,215 @@ def wait_for_api() -> None:
+        typer.Option(
+            "--document-filesystem-handler",
+            metavar="DOCUMENT_FILESYSTEM_HANDLER",
+            default_factory=lambda: "ragna.core.LocalDocument",
+            help="Select source for storing vector embeddings.",
+        ),
+    ],
+    database_url: Annotated[
+        str,
+        typer.Option(
+            "--database-url",
+            metavar="DATABASE_URL",
+            default_factory=lambda: f"sqlite:///{local_root()}/ragna.db",
+            help="URL for ragna database.",
+        ),
+    ],
+    corpus_name: Annotated[
+        Optional[str],
+        typer.Option(
+            "--corpus-name",
+            metavar="CORPUS_NAME",
+            default_factory=lambda: None,
+            help="Name of the corpus to ingest the documents into.",
+        ),
+    ],
+    documents: list[Path],
+    verbose: Annotated[bool, typer.Option("--verbose")] = False,
+) -> None:
+    try:
+        source_storage_class = SourceStorageImport(
+            source_storage_class=source_storage
+        ).source_storage_class
+    except ValidationError:
+        raise typer.BadParameter(f"Unable to import source storage: {source_storage}")
+
+    try:
+        document_filesystem_handler_class = DocumentFilesystemHandlerImport(
+            document_filesystem_handler_class=document_filesystem_handler
+        ).document_filesystem_handler_class
+    except ValidationError:
+        raise typer.BadParameter(
+            f"Unable to import document filesystem handler: {document_filesystem_handler}"
+        )
+
+    try:
+        getattr(document_filesystem_handler_class, "from_path")
+    except AttributeError:
+        raise typer.BadParameter(
+            f"Document filesystem handler {document_filesystem_handler} does not support ingesting documents. Please implement a `from_path` method."
+        )
+
+    make_session = database.get_sessionmaker(database_url)
+
+    BATCH_SIZE = 10
+    bad_documents = []
+    with tqdm(total=len(documents) // BATCH_SIZE) as pbar:
+        for batch_number in range(0, len(documents), BATCH_SIZE):
+            document_collection = []
+            document_metadata_collection = []
+
+            for document in documents[batch_number : batch_number + BATCH_SIZE]:
+                try:
+                    doc_instance = document_filesystem_handler_class.from_path(document)
+                    document_collection.append(doc_instance)
+                    document_metadata_collection.append(
+                        (
+                            schemas.Document.from_core(doc_instance),
+                            doc_instance.metadata,
+                        )
+                    )
+                except Exception:
+                    bad_documents.append(document)
+
+            if not document_collection:
+                continue
+
+            # TODO: add a check to see if the documents already exist in the database
```

Comment:
Flagging a TODO

Reply:
I'v added a short-term checkpointing solution that stores all paths in a local file.